pkg/operator/operator.go

```diff
@@ -162,7 +153,7 @@ func (c *csiSnapshotOperator) sync() error {
 	syncErr := c.handleSync(instanceCopy)
 	c.updateSyncError(&instanceCopy.Status.OperatorStatus, syncErr)
 
-	if _, _, err := v1helpers.UpdateStatus(c.client, func(status *operatorv1.OperatorStatus) error {
+	if _, _, err := v1helpers.UpdateStatus(context.TODO(), c.client, func(status *operatorv1.OperatorStatus) error {
```

Comment:
Can you plumb through the `ctx` variable from `starter.go: RunOperator()`? You could start by changing the signature of the `Run()` function to `Run(ctx context.Context, workers int)`.

Reply:
Sounds great, modified the code to use context